WalletWasabi.Fluent/ViewModels/Wallets/WalletViewModel.cs

```diff
@@ -115,6 +115,20 @@ args switch
 		this.WhenAnyValue(x => x.IsWalletBalanceZero)
 			.Subscribe(_ => IsSendButtonVisible = !IsWalletBalanceZero && (!wallet.KeyManager.IsWatchOnly || wallet.KeyManager.IsHardwareWallet));
 
+		this.WhenAnyValue(x => x.IsWalletBalanceZero)
+			.Do(empty =>
+			{
+				if (wallet.KeyManager.IsWatchOnly)
+				{
+					return;
+				}
+
+				Tiles = empty ? TileHelper.GetWatchOnlyWalletTiles(this)
+					: TileHelper.GetNormalWalletTiles(this);
```

Comment:
nACK.

This looks quick and dirty, and surely works and does the expected thing **now**. But it is not very future proof IMO.

We're mixing two unrelated concepts: watch-only wallet and the wallet balance. This code is prone to break if watch-only wallet-related behavior changes in the future. For instance by adding another tile that only works for watch-only, but then you will wrongly see it in your normal wallet with zero balance, because of this code

Also, there is unnecessary indirection here. Meaning the logic resides in an upper "layer" (the `WalletViewModel`) instead of residing in the Tile-related classes. This makes the code harder to maintain because if someone introduces new tiles or behavior changes they will surely modify tile-related classes and not `WalletViewModel`.

I suggest moving this logic into `TileHelper` and refactor the code in a way that does not depend on unrelated factors such as `IsWatchOnly`

Reply:
Makes sense, im trying the TileHelper way now